common/services/user-locations.js

```diff
@@ -0,0 +1,68 @@
+const axios = require('axios')
+
+const { AUTH_PROVIDERS, NOMIS_ELITE2_API } = require('../../config')
+const { decodeAccessToken } = require('../lib/access-token')
+const referenceDataService = require('./reference-data')
+
+async function getUserLocations(token) {
+  var agencies
+
+  const tokenData = decodeAccessToken(token)
+
+  switch (tokenData.auth_source) {
+    case 'auth':
+      agencies = await _getAuthAgencies(token)
+      break
+    case 'nomis':
+      agencies = await _getNomisAgencies(token)
+      break
+  }
+
+  const locations = await Promise.all(
+    agencies.map(async agencyId => {
+      return referenceDataService.getLocations({ nomisAgencyId: agencyId })
+    })
+  )
+
+  return locations.map(results => {
+    return results[0].id
+  })
+}
+
+async function _getAuthAgencies(token) {
+  const tokenData = decodeAccessToken(token)
+
+  const response = await axios.get(
+    AUTH_PROVIDERS.hmpps.groups_url(tokenData.user_name),
+    {},
+    _authHeaders(token)
+  )
+
+  return response.data.map(group => {
+    return group.groupCode.replace(/^PECS_/, '')
+  })
+}
+
+async function _getNomisAgencies(token) {
+  const response = await axios.get(
+    NOMIS_ELITE2_API.user_caseloads_url,
+    {},
+    _authHeaders(token)
+  )
+
+  return response.data.map(caseLoad => {
```

Comment:
Could be shortened to:

```js
return response.data.map(caseLoad => caseLoad.caseLoadId)
```

or, if you use [lodash's map](https://lodash.com/docs/4.17.15#map) function it could be:

```js
return map(response.data, 'caseLoadId')
```